### PR TITLE
[Cucumber] [Cucumber/docs]Selenium webdriver.md and watir webdriver

### PR DESCRIPTION
**Pull request by mgundala**
## Summary

Added Selenium-webdriver.md and watir-webdriver.md files to cucumber docs

## Details

There were few conflicts in previous documentation (Pull request #54). So created a new pull request.

## Motivation and Context

Adding more value to documentation. Missed it for a while will continue going forward

## How Has This Been Tested?

Need to be reviewed by peers

## Screenshots (if appropriate):
NA
## Types of changes
NA
- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/selenium-webdriver.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+## Selenium WebDriver
+
+WebDriver is designed to provide a simpler, more concise programming interface in addition to addressing some limitations in the Selenium-RC API. Selenium-WebDriver was developed to better support dynamic web pages where elements of a page may change without the page itself being reloaded. WebDriver�s goal is to supply a well-designed object-oriented API that provides improved support for modern advanced web-app testing problems.
+ 
+Let us see an example how cucumber can fit into selenium-webdriver in UI testinng, by converting [Selenium-Web driver by example     ](http://docs.seleniumhq.org/docs/03_webdriver.jsp#introducing-the-selenium-webdriver-api-by-example "Selenium-Webdriver by Example tutorial")
```
Comment (olleolleolle) [inline docs/selenium-webdriver.md]: Typo `testinng` => `testing`

Comment (olleolleolle) [inline docs/selenium-webdriver.md]: Same line: the link seems to have a few spaces after `example`.

File: SUMMARY.md [hunk 2]
```diff
@@ -50,7 +50,8 @@
     * [Shallow Tests](./docs/404.md)
     * [Narrow Tests](./docs/404.md)
     * [UI Testing](./docs/404.md)
-      * [Selenium WebDriver](./docs/404.md)
+      * [Selenium WebDriver](/docs/selenium-webdriver.md)
+	  * [Watir WebDriver](/docs/watir-webdriver.md)
```
Comment (olleolleolle) [inline SUMMARY.md]: Q: is Watir article meant to be a sub-bullet-point of _Selenium WebDriver_ or of _UI Testing_?

File: docs/selenium-webdriver.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ 
+ ```java
+ 
+ USING JAVA
+ {% codetabs name="Java", type="java" -%}
+ package class.exmple;
+ 
+ public class ExampleSteps {
+ 
+      private final WebDriver driver = new FirefoxDriver();
```
Comment (olleolleolle) [inline docs/selenium-webdriver.md]: The formatting of the `ExampleSteps` Java class is not consistent.

File: docs/selenium-webdriver.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      @Given("^I am on the Google search page$"\)
+      public void I_visit_google() {
+      driver.get("https:\\www.google.com");
```
Comment (olleolleolle) [inline docs/selenium-webdriver.md]: The URL's slashes are the wrong kind. `\\` should be `//`.

File: docs/selenium-webdriver.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+ ```
+ 
+```ruby
+ USING RUBY:
+ {% codetabs name="Ruby", type="rb" -%}
+ require 'rubygems'
+ require 'selenium-webdriver'
+ 
+ Given("^I am on the Google search page$") do
+  driver = Selenium::WebDriver.for :firefox
```
Comment (olleolleolle) [inline docs/selenium-webdriver.md]: For Ruby, use 2 spaces as the indentation inside `do..end` blocks.

File: docs/selenium-webdriver.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      private final WebDriver driver = new FirefoxDriver();
+      @Given("^I am on the Google search page$"\)
+      public void I_visit_google() {
+      driver.get("https:\\www.google.com");
+     }
+ 
+     @When("^I search for \"(.*)\"$")
+       public void search_for(String query) {
+          WebElement element = browser.findElelment(By.name("q"));
+          \\Enter Something to search for
```
Comment (olleolleolle) [inline docs/selenium-webdriver.md]: The Java code comment symbol is `//`, not `\\`.

File: docs/watir-webdriver.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
+##Watir Webdriver
```
Comment (olleolleolle) [inline docs/watir-webdriver.md]: Use a space to separate the `##` from the text of the heading.

File: docs/watir-webdriver.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+Watir, pronounced water, is an open-source (BSD) family of Ruby libraries for automating web browsers. It allows you to write tests that are easy to read and maintain. It is simple and flexible.
+ 
+Watir drives browsers the same way people do. It clicks links, fills in forms, presses buttons. Watir also checks results, such as whether expected text appears on the page.
+ 
+Watir is a family of Ruby libraries but it supports your app no matter what technology it is developed in. While Watir supports only Internet Explorer on Windows, Here comes Watir-WebDriver to solve single browser testing and support Chrome, Firefox, Internet Explorer, Opera and also running in headless mode (HTMLUnit).
+ 
+ Now without wasting anytime quickly jump in to a sample UI testing program using Watir
```
Comment (olleolleolle) [inline docs/watir-webdriver.md]: Typo: `anytime` => `any time`

Comment (olleolleolle) [inline docs/watir-webdriver.md]: The sentence should end with a period (`.`).

File: docs/watir-webdriver.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+ 
+ ```ruby
+ {% codetabs name="Ruby", type="rb" -%}
+ require "rubygems"
+ require "rspec"
+ require "watir-webdriver"
+  
+ describe "google.com" do
+   let(:browser) { @browser ||= Watir::Browser.new :firefox }
```
Comment (olleolleolle) [inline docs/watir-webdriver.md]: Can this RSpec `let` be rephrased as

```ruby
let(:browser) { Watir::Browser.new(:firefox) }
```

File: SUMMARY.md (hunk 2 above)
Comment (mgundala) [inline SUMMARY.md]: it's a sub content of UI testing

File: docs/watir-webdriver.md (hunk 9 above)
Comment (mgundala) [inline docs/watir-webdriver.md]: We can

**Comment by mgundala** [comment]
Hi Jonnson,

Those are valuable suggestions. Thanks for ur time in reviewing this.
My intention is to have a basic knowledge of web driver and how these can be used along with Cucumber and add value to the tests.

Please correct me if I am going down a wrong path

File: SUMMARY.md (hunk 2 above)
Comment (olleolleolle) [inline SUMMARY.md]: (Good, then it should be at the same level of indent as Selenium WebDriver.)

Comment (mgundala) [inline SUMMARY.md]: Not sure why it is showing inline content

**Comment by olleolleolle** [comment]
Did I discourage you with the "big red symbol"? I hope not!

The red symbol only means "I'd like to see some of these changes added". It means that I'd like for you to work _more_ on this thing, that it's important, and that it can be made suuuper-useful.

**Comment by gmk1217** [comment]
No, Not at all.

Every suggestion you made is valuable.
Making necessary corrections as per your suggestions.
According to me, More the corrections more the value it is.(little bit
selfish though.. :) )
Will Create a new pull request, to reduce confusion I closed the previous.


---

Thanks,

Mohan G

**Comment by brasmusson** [comment]
> Will Create a new pull request, to reduce confusion I closed the previous.

@gmk1217 Does not creating a new pull request just increase the confusion? The discussion and comment here will not be accessible from the new pull request. As pull requests on Github is as flexible as they are, it is possible to add commits to them, remove commits from them, rebase them, squash the commits on them, all by manipulating the branch they are created from, I think it is better not to close a pull request, just to open another for the same change.

**Pull request closed without merging**